ccl/partitionccl: TestInitialPartitioning/all_indexes_-_shuffled failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/9376ca50f874e475189b1fe1f7ef8f4e0f6c18b4

Parameters:
```
TAGS=
GOFLAGS=
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=658828&tab=buildLog

```
    	partition_test.go:1145: error executing 'ALTER INDEX "all indexes"@idx1 EXPERIMENTAL CONFIGURE ZONE 'constraints: [+n2]';ALTER INDEX "all indexes"@idx2 EXPERIMENTAL CONFIGURE ZONE 'constraints: [+n3]';': pq: constraint "+n3" matches no existing nodes within the cluster - did you enter it correctly?
```

--- a/pkg/ccl/partitionccl/partition_test.go
+++ b/pkg/ccl/partitionccl/partition_test.go
@@ -23,6 +23,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/config"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/parser"
@@ -1117,6 +1118,31 @@ func setupPartitioningTestCluster(ctx context.Context, t testing.TB) (*sqlutils.
 	// Disabling store throttling vastly speeds up rebalancing.
 	sqlDB.Exec(t, `SET CLUSTER SETTING server.declined_reservation_timeout = '0s'`)
 	sqlDB.Exec(t, `SET CLUSTER SETTING server.failed_reservation_timeout = '0s'`)
+
+	// Make sure all stores are present in the NodeStatus endpoint or else zone
+	// config changes may flake (#25488).
+	testutils.SucceedsSoon(t, func() error {
+		url := tc.Server(0).ServingAddr()
+		conn, err := tc.Server(0).RPCContext().GRPCDial(url).Connect(context.Background())
+		if err != nil {
+			return err
+		}
+		client := serverpb.NewStatusClient(conn)
+		response, err := client.Nodes(context.Background(), &serverpb.NodesRequest{})
+		if err != nil {
+			return err
+		}
+
+		if len(response.Nodes) != 3 {
+			return fmt.Errorf("not enough nodes registered: %+v", response)
+		}
+		for _, node := range response.Nodes {
+			if len(node.StoreStatuses) != 1 {
+				return fmt.Errorf("expected 1 StoreStatus in NodeStatus, got %+v", node)
+			}
+		}
+		return nil
+	})
 
 	return sqlDB, func() {
 		tc.Stopper().Stop(context.Background())

